src/lib/mina.ts

```diff
@@ -129,9 +130,204 @@ type Transaction = {
   /**
    * Sends the {@link Transaction} to the network.
    */
-  send(): Promise<TransactionId>;
+  send(): Promise<PendingTransaction>;
+
+  /**
+   * Sends the {@link Transaction} to the network, unlike the standard send(), this function will throw an error if internal errors are detected.
+   */
+  sendOrThrowIfError(): Promise<PendingTransaction>;
+};
+
+/**
+ * Represents a transaction that has been submitted to the blockchain but has not yet reached a final state.
+ * The `PendingTransaction` type extends certain functionalities from the base `Transaction` type,
+ * adding methods to monitor the transaction's progress towards being finalized (either included in a block or rejected).
+ */
+type PendingTransaction = Pick<
```

Comment:
This file is starting to get big :O Should we separate transaction-related logic into its own module and let `mina.ts` just be responsible for the interaction/implementations between `LocalBlockchain` and `Network`?

Reply:
As a first step I'd actually move `LocalBlockchain` to a separate file, since it's kind of a testing utility and not part of the "main" API

maybe into `mina/local-blockchain.ts`

Reply:
and yeah, transaction construction e.g. `createTransaction` and the `Transaction` interface could be another separate file

Reply:
done!